### PR TITLE
Enforce 'go test' in ZTP PolicyGenerator code

### PR DESCRIPTION
**Pull request by lack**
- ztp: add hack script to run 'go test' with the right environment
- infra: Enforce the ztp PolicyGenerator 'go test' passes in ci-job

### CODE REVIEW

**Review by serngawy — APPROVED** [review]

File: cnf-tests/hack/govet.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+. $(dirname "$0")/common.sh
```
Comment (fedepaol) [inline cnf-tests/hack/govet.sh:3]: Would it make sense to have one single govet.sh file in the root `hack` folder instead of two?

Comment (lack) [reply cnf-tests/hack/govet.sh:3]: I'm thinking I might actually redistribute this code a bit - In order to have to not involve 'infra' every time we want to make a change within ztp, I was thinking I'd change the top-level `make job-ci` call `$(MAKE) -c ztp job-ci`, which can then be adapted to do 'go vet' and 'go test' and whatever other checks we want to add in the future.

How do you feel about that?

**Comment by lack** [comment]
Closing this one in favor of #702

**Pull request closed without merging**